tests/plugin/test_http_proxy_plugins.py

```diff
@@ -24,21 +25,23 @@
 from proxy.http.proxy import HttpProxyPlugin
 from proxy.common.utils import build_http_request, bytes_, build_http_response
 from proxy.common.constants import PROXY_AGENT_HEADER_VALUE, DEFAULT_HTTP_PORT
-
 from proxy.plugin import ProposedRestApiPlugin, RedirectToCustomServerPlugin
 
 from .utils import get_plugin_by_test_name
 
+from ..test_assertions import Assertions
 
-class TestHttpProxyPluginExamples(unittest.TestCase):
 
-    @mock.patch('selectors.DefaultSelector')
-    @mock.patch('socket.fromfd')
-    def setUp(
-        self,
-        mock_fromfd: mock.Mock,
-        mock_selector: mock.Mock,
-    ) -> None:
+class TestHttpProxyPluginExamples(Assertions):
+
+    @pytest.fixture(autouse=True)   # type: ignore[misc]
+    def _setUp(self, request: Any, mocker: MockerFixture) -> None:
+        self.mock_fromfd = mocker.patch('socket.fromfd')
```

Comment:
Instead of trying to have side effects with unclear influence targets on the class instance, make a fixture that constructs and returns this. In fact, make a cascade of pure-function based fixtures that are easy to compose and redefine at multiple levels.

Reply:
Totally, I see what you mean.  Ideally fixtures should have returned the result back to be used within test.  But at this point, I didn't want to disrupt the entire test suite.  I had to use `pytest` because I needed `asyncio` support, as core was updated to use `async` almost everywhere.

I'll come back to tests and nail them properly.  With `pytest` we can also reduce a lot of redundant code in tests.  When I revisited them today, I could see the redundancy :).  But left it for later.

Reply:
Tracking here https://github.com/abhinavsingh/proxy.py/issues/775